test/test_base_service.py

```diff
@@ -810,56 +811,71 @@ def test_json():
     service = AnyServiceV1('2018-11-20', authenticator=NoAuthAuthenticator())
     req = service.prepare_request('POST', url='', headers={
                                   'X-opt-out': True}, data={'hello': 'world', 'fóó': 'bår'})
-    assert req.get('data') == b'{"hello": "world", "f\\u00f3\\u00f3": "b\\u00e5r"}'
+    assert req.get(
+        'data') == b'{"hello": "world", "f\\u00f3\\u00f3": "b\\u00e5r"}'
 
 
-def test_trailing_slash():
-    assert strip_extra_slashes('') == ''
-    assert strip_extra_slashes('//') == '/'
-    assert strip_extra_slashes('/////') == '/'
-    assert strip_extra_slashes('https://host') == 'https://host'
-    assert strip_extra_slashes('https://host/') == 'https://host/'
-    assert strip_extra_slashes('https://host//') == 'https://host/'
-    assert strip_extra_slashes('https://host/path') == 'https://host/path'
-    assert strip_extra_slashes('https://host/path/') == 'https://host/path/'
-    assert strip_extra_slashes('https://host/path//') == 'https://host/path/'
-    assert strip_extra_slashes('https://host//path//') == 'https://host//path/'
-
+def test_service_url_handling():
     service = AnyServiceV1(
-        '2018-11-20', service_url='https://host/', authenticator=NoAuthAuthenticator())
-    assert service.service_url == 'https://host/'
+        '2018-11-20', service_url='https://host///////', authenticator=NoAuthAuthenticator())
+    assert service.service_url == 'https://host'
+
     service.set_service_url('https://host/')
-    assert service.service_url == 'https://host/'
+    assert service.service_url == 'https://host'
+
     req = service.prepare_request('POST',
                                   url='/path/',
                                   headers={'X-opt-out': True},
                                   data={'hello': 'world'})
-    assert req.get('url') == 'https://host//path/'
```

Comment:
we can no longer have double-slashes between the service_url and the operation path